### PR TITLE
Test against all supported runtime

### PR DESCRIPTION
**Pull request by billy1624**
Related to #86

### CODE REVIEW

**Comment by tyt2y3** [comment]
Brilliant. A macro ?

File: src/runtime.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#[cfg(any(
+    feature = "runtime-async-std",
+    all(
```
Comment (tyt2y3) [inline src/runtime.rs:3]: This any all combo is a little strange?

**Comment by billy1624** [comment]
Macro! :P

File: src/runtime.rs (hunk 1 above)
Comment (billy1624) [reply src/runtime.rs:3]: To support simple `cargo test` without specifying any runtime preference

Comment (billy1624) [reply src/runtime.rs:3]: Default use async-std

Comment (tyt2y3) [reply src/runtime.rs:3]: I still don't understand why is it
```
async-std OR (!async-std AND !actix AND !tokio)
```
if any runtime is specified, then we use that runtime.
if no runtime is specified, then we simply do nothing?

**Comment by tyt2y3** [comment]
As per our discussion, I think we'd try:

Using `smol` async runtime for unit & doc tests

Require at least one of the production runtimes available for running integration tests.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -174,27 +207,31 @@ jobs:
...
     name: Postgres
     runs-on: ubuntu-20.04
+    needs: check
     env:
       DATABASE_URL: "postgres://root:root@localhost"
     strategy:
       matrix:
-        version: [13.3, 12.7, 11.12, 10.17, 9.6.22]
-        runtime: [async-std-native-tls]
+        version: [13.3, 12.7]
```
Comment (tyt2y3) [inline .github/workflows/rust.yml:232]: We could not support old versions of Postgres?

Comment (billy1624) [reply .github/workflows/rust.yml:232]: We can just thinking if it's too much loll

Comment (billy1624) [reply .github/workflows/rust.yml:232]: Let me add it back then

**Comment by billy1624** [comment]
Btw... for Postgres integration tests, this happened from time to time...

> thread 'left_join' panicked at 'could not insert baker: Query("error returned from database: column \"bakery_id\" is of type integer but expression is of type boolean")', tests/relational_tests.rs:51:6

https://github.com/SeaQL/sea-orm/runs/3362229412?check_suite_focus=true#step:6:107

**Comment by billy1624** [comment]
Correct! 👻

**Comment by tyt2y3** [comment]
> > Btw... for Postgres integration tests, this happened from time to time...
> > > thread 'left_join' panicked at 'could not insert baker: Query("error returned from database: column "bakery_id" is of type integer but expression is of type boolean")', tests/relational_tests.rs:51:6
> > 
> > 
> > https://github.com/SeaQL/sea-orm/runs/3362229412?check_suite_focus=true#step:6:107
> 
> 💩 from time to time? there is a cause of non-determinism?

Well. That's bad. @samsamai would you by chance has some guess?

**Comment by billy1624** [comment]
All these suffer the same error loll
- https://github.com/SeaQL/sea-orm/runs/3367814059?check_suite_focus=true#step:6:106
- https://github.com/SeaQL/sea-orm/runs/3362229412?check_suite_focus=true#step:6:107
- https://github.com/SeaQL/sea-orm/runs/3361018305?check_suite_focus=true#step:7:107
- https://github.com/SeaQL/sea-orm/runs/3360297887?check_suite_focus=true#step:7:107

**Comment by tyt2y3** [comment]
I have a remote idea
https://github.com/SeaQL/sea-query/blob/b384d4d26aea0ae2ef1bc112558d2cd07a798544/src/driver/sqlx_postgres.rs#L7

**Comment by tyt2y3** [comment]
I have a theory now. The prepared statement is exectued elsewhere with an integer value, so when it's executed 2nd time, the protocol would complain type mismatch.

**Comment by tyt2y3** [comment]
https://www.postgresql.org/docs/13/protocol-message-formats.html

**Comment by billy1624** [comment]
We can merge this PR frist? As this issue is solely related to sea-query.

**Pull request merged**